src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -57,26 +62,20 @@
     public StaticHostProvider(Collection<InetSocketAddress> serverAddresses)
             throws UnknownHostException {
         for (InetSocketAddress address : serverAddresses) {
-            InetAddress ia = address.getAddress();
-            InetAddress resolvedAddresses[] = InetAddress.getAllByName((ia!=null) ? ia.getHostAddress():
-                address.getHostName());
+            InetAddress resolvedAddresses[];
+            try {
+                Method m = InetSocketAddress.class.getDeclaredMethod("getHostString");
+                m.setAccessible(true);
+                resolvedAddresses = InetAddress.getAllByName((String) m.invoke(address));
+            } catch (IllegalAccessException e) {
+                resolvedAddresses = InetAddress.getAllByName(getHostString(address));
+            } catch (NoSuchMethodException e) {
+                resolvedAddresses = InetAddress.getAllByName(getHostString(address));
+            } catch (InvocationTargetException e) {
+                resolvedAddresses = InetAddress.getAllByName(getHostString(address));
+            }
```

Comment:
The signature of getDeclaredMethod said it also could throw SecurityException, not sure if we should catch it or not here.